concent_api/api_testing_helpers.py

```diff
@@ -73,7 +74,15 @@ def api_request(host, endpoint, private_key, public_key, data = None, headers =
         )
         print('STATUS: {} {}'.format(response.status_code, http.client.responses[response.status_code]))
         print('MESSAGE:')
-        print_golem_message(decoded_response, private_key, public_key)
+        if response.headers['Content-Type'] == 'application/octet-stream':
+            print_golem_message(decoded_response, private_key, public_key)
```

Comment:
You're not handling the case where `decoded_response` is `None`.

You should actually move the `load()` call here.